tests/contenttypes_tests/tests.py

```diff
@@ -113,6 +114,21 @@ class Meta:
         self.assertEqual(ct.model, 'modelcreatedonthefly')
         self.assertEqual(force_text(ct), 'modelcreatedonthefly')
 
+    def test_shortcut_view_with_null_site_fk(self):
+        """
+        The shortcut view works if a model's ForeignKey to site is None.
+        """
+        request = HttpRequest()
+        request.META = {
+            "SERVER_NAME": "Example.com",
+            "SERVER_PORT": "80",
+        }
+        obj = ModelWithNullFKToSite.objects.create(title="object title")
+        short_url = '/shortcut/%s/%s/' % (ContentType.objects.get_for_model(ModelWithNullFKToSite).id, obj.pk)
+        response = self.client.get(short_url)
+        self.assertRedirects(response, 'http://testserver%s' % obj.get_absolute_url(),
```

Comment:
prefer hanging indent, also `status_code=302` is unneeded since it's the default:
```python
self.assertRedirects(
    response, 'http://testserver%s' % obj.get_absolute_url(),
    fetch_redirect_response=False,
)
```